packages/desktop-client/src/components/reports/graphs/sankey-spreadsheet.tsx

```diff
@@ -0,0 +1,179 @@
+      try {
+        return await Promise.all(
+          categories.map(async mainCategory => {
+            let subcategoryBalances = await Promise.all(
+              mainCategory.categories
+                .filter(subcategory => subcategory.is_income !== 1)
+                .map(async subcategory => {
+                  const results = await runQuery(
+                    q('transactions')
+                      .filter({
+                        [conditionsOpKey]: filters,
+                      })
+                      .filter({
+                        $and: [
+                          { date: { $gte: start + '-01' } },
+                          { date: { $lte: end + '-31' } },
+                        ],
+                      })
+                      .filter({ category: subcategory.id })
+                      .calculate({ $sum: '$amount' }),
+                  );
+                  return {
+                    subcategory: subcategory.name,
+                    value: results.data * -1,
+                  };
+                }),
+            );
+
+            // Here you could combine, reduce or transform the subcategoryBalances if needed
+            return {
+              name: mainCategory.name,
+              balances: subcategoryBalances,
+            };
+          }),
+        );
+      } catch (error) {
+        console.error('Error fetching category data:', error);
+        throw error; // Re-throw if you want the error to propagate
+      }
+    }
+
+    // retrieve all income subcategory payees
+    async function fetchIncomeData() {
+      // Map over allIncomeSubcategories and return an array of promises
+      const promises = allIncomeSubcategories.map(subcategory => {
+        return runQuery(
+          q('transactions')
+            .filter({
+              [conditionsOpKey]: filters,
+            })
+            .filter({
+              $and: [
+                { date: { $gte: start + '-01' } },
+                { date: { $lte: end + '-31' } },
+              ],
+            })
+            .filter({ category: subcategory.id })
+            .groupBy(['payee'])
+            .select(['payee', { amount: { $sum: '$amount' } }]),
+        );
+      });
+
+      // Use Promise.all() to wait for all queries to complete
+      const resultsArrays = await Promise.all(promises);
+
+      // unravel the results
+      let payeesDict = {};
+      resultsArrays.forEach(item => {
+        item.data.forEach(innerItem => {
+          payeesDict[innerItem.payee] = innerItem.amount;
+        });
+      });
+
+      // Fetching names based on payee ID
+      let payeeNames = {};
+      for (let id in payeesDict) {
+        const result = await runQuery(
+          q('payees').filter({ id: id }).select(['name']),
+        );
+        if (result.data.length > 0 && result.data[0].name) {
+          payeeNames[result.data[0].name] = payeesDict[id];
+        }
+      }
+      return payeeNames;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could we make a single query here with multiple IDs and then convert the resulting array to a payee-name-map that you need? That would be much more optimal.